modules/arm_plugin/src/arm_converter/arm_converter_region_yolo.cpp

```diff
@@ -19,9 +19,10 @@ template<> Converter::Conversion::Ptr Converter::Convert(const opset::RegionYolo
                                     node.get_mask());
     };
 
-    if (node.get_input_element_type(0) != ngraph::element::f32) {
-        THROW_IE_EXCEPTION << "Unsupported Type: " << node.get_input_element_type(0);
+    switch (node.get_input_element_type(0)) {
+        case ngraph::element::Type_t::f16 : return make(ngraph::runtime::reference::region_yolo<ngraph::float16>);
```

Comment:
@apankratovantonp should not we call these references under `#ifdef __ARM_FEATURE_FP16_VECTOR_ARITHMETIC`? They seem to be not needed for FP32 execution

Reply:
And maybe in opposite case - FP32 kernels are not needed for FP16 case..

Reply:
I preferer to leave binary size optimization to selective build feature and enable it for all types.

Reply:
You will have dead code. E.g. FP16 kernels physically cannot be run on FP32 arch.